extensions/cl_ext_relaxed_printf_string_address_space.asciidoc

```diff
@@ -0,0 +1,123 @@
+cl_ext_relaxed_printf_address_space
+====================================
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:blank: pass:[ +]
+:data-uri:
+:icons: font
+include::../config/attribs.txt[]
+:source-highlighter: coderay
+
+== Name Strings
+
++cl_ext_relaxed_printf_address_space+
+
+== Contact
+
+Pekka Jääskeläinen, Parmance (pekka /at/ parmance /dot/ com)
+
+== Contributors
+
+Pekka Jääskeläinen, Parmance +
+Henry Linjamäki, Parmance +
+Brice Videau, Argonne National Laboratory +
+Anastasia Stulova, Arm
+
+== Notice
+
+Copyright (c) 2022 Parmance and Argonne National Laboratory.
+
+== Status
+
+Early Draft
+
+== Version
+
+Built On: {docdate} +
+Revision: 3
+
+== Dependencies
+
+This extension is written against the OpenCL Specification Version 1.2, Revision 19 and
+the OpenCL C 2.0 Specification, Revision 29.
+
+The extension works with OpenCL 1.2 for the overloads that do not require generic address
+space support, and requires at least OpenCL 2.0 for the generic address space one.
+
+== Overview
+
+The printf as originally defined in OpenCL 1.2 deviates from the C99 printf
+enough that makes using it as a drop-in function for programs originally
+written in C99 that use printf sometimes cumbersome.
+
+This extension addresses a major gap in this regard by allowing the format
+string specifier and the arguments to the %s conversion specifier to originate
+from any address space, not only from the constant address space as specified
+in the OpenCL 1.2 specification.
+
+This relaxation implies that the format strings and the strings passed as %s
+arguments can be dynamically produced during the execution time of the kernel.
+
+== Modifications to the OpenCL C 1.2 Specification
+
+=== Modify Table 6.21
+
+Add new function overloads (added to the bottom left cell of Table 6.21): ::
++
+
+[source,c]
+----
+int printf(global char * restrict format, ...)
+int printf(local char * restrict format, ...)
+int printf(private char * restrict format, ...)
```

Comment:
Should we support the generic address space as well if the generic address space is supported ("OpenCL C 2.0 or OpenCL C 3.0 with the __opencl_c_generic_address_space feature")?

Reply:
Yes, it's the second part of the specs: https://github.com/KhronosGroup/OpenCL-Docs/pull/807/files/e95c2be61869a7852c14187edeceee5d3bdec36a#diff-807af2fed1c59e28e26fe56cd246632c5f8d5f809a3dea8a005859b941bc16f5R89